CMakeLists.txt

```diff
@@ -17,6 +17,10 @@ cmake_dependent_option(WITH_GLOG "Use glog for logging." ON
                        "WITH_GFLAGS" OFF)
 add_feature_info(GLOG WITH_GLOG "provides logging configurability.")
 
+option(BUILD_SHARED_LIBS "Build shared libraries instead of static." ON)
```

Comment:
Add comment that this is OFF for Windows (if that's indeed the case -- not sure if you're adding it for Windows, or just for consistency with normal CMake practice).

Reply:
It is on for all platforms by default. The goal is to allow a simple flag to enable static builds instead of requiring a source change (as they were hardcoded SHARED) before. Will separate into another PR.